alert/interface.go

```diff
@@ -0,0 +1,48 @@
+package alert
+
+import "time"
+
+const (
+	NoAlertFound Error = "no alert found for the input key"
+)
+
+// Error implements Error interface
+type Error string
+
+// Error reports underlying error as a string
+func (e Error) Error() string {
+	return string(e)
+}
+
+// Altertable defines an interface that can be managed using Alerter interface.
+// An instance of Alertable is an alert object
+type Alertable interface {
+	// Type indicates alert type
+	Type() int64
+	// ID is the resource ID of alert generator
+	ID() string
+	// Reason reports a brief comment on why alert was raised
+	Reason() string
+	// FirstSeen is the time stamp of first alert of this type and ID
+	FirstSeen() time.Time
+	// LastSeen is the time stamp of last alert of this type and ID
+	LastSeen() time.Time
+	// Marshal defines serialization of implementing object
+	Marshal() ([]byte, error)
+	// Unmarshal defines deserialization of implementing object
+	Unmarshal(b []byte) error
+}
+
+// Alerter defines an interface to manage alerts
+type Alerter interface {
```

Comment:
nit: I would vote for AlertManager as the name instead of Alerter. Still your call.

Reply:
yeah, let's make this another manager :)